src/core/text/TextMetrics.js

```diff
@@ -255,6 +284,109 @@ export default class TextMetrics
         return width;
     }
 
+    /**
+     * trims breaking whitespaces from string
+     *
+     * @param  {string}  text  The text
+     * @return {string}  trimmed string
+     */
+    static trimRight(text)
+    {
+        for (let i = text.length - 1; i >= 0; i--)
+        {
+            const char = text[i];
+
+            if (!TextMetrics.isBreakingSpace(char))
+            {
+                break;
+            }
+
+            text = text.slice(0, -1);
+        }
+
+        return text;
+    }
+
+    /**
+     * Determines if char is a newline.
+     *
+     * @param  {string}  char  The character
+     * @return {boolean}  True if newline, False otherwise.
+     */
+    static isNewline(char)
+    {
+        const newlines = [
+            0x000A, // line feed
+            0x000D, // carriage return
+        ];
+
+        return (newlines.indexOf(char.charCodeAt(0)) >= 0);
+    }
+
+    /**
+     * Determines if char is a breaking whitespace.
+     *
+     * @param  {string}  char  The character
+     * @return {boolean}  True if whitespace, False otherwise.
+     */
+    static isBreakingSpace(char)
```

Comment:
And a very basic unit test for this. Make it return true/false, check for return type

Reply:
will do

Reply:
Done